src/Compat.php

```diff
@@ -2684,6 +2684,29 @@ public static function polyfill_is_fast()
         return PHP_INT_SIZE === 8;
     }
 
+    /**
+     * @param int $iterations Number of multiplications to attempt
+     * @param int $maxTimeout Milliseconds
+     * @return bool           TRUE if we're fast enough, FALSE is not
+     * @throws SodiumException
+     */
+    public static function runtime_speed_test($iterations, $maxTimeout)
+    {
+        if (self::polyfill_is_fast()) {
+            return true;
+        }
+        $end = 0.0;
+        $start = microtime(true);
+        $a = ParagonIE_Sodium_Core32_Int64::fromInt(random_int(3, 1 << 16));
+        for ($i = 0; $i < $iterations; ++$i) {
+            $b = ParagonIE_Sodium_Core32_Int64::fromInt(random_int(3, 1 << 16));
+            $a->mulInt64($b);
+        }
```

Comment:
Although this isn't a problem right now, FYI this might have a problem in the future with PHP's Dead-Code Elimination optimiser. I've looked at the optimised opcodes at present and it seems OK for now.

See https://www.youtube.com/watch?v=WJJKZM8bruQ#t=28m10s & http://talks.php.net/etsy18#/dce for some examples.